secret_santa.exs

```diff
@@ -0,0 +1,35 @@
+# The code is largely inspired by this tweet: https://twitter.com/gausby/status/1200462062733987841
+
+defmodule EnumUtils do
+  def rotate_left([h | t]) do
+    t ++ [h]
+  end
+end
+
+defmodule SecretSanta do
+  def pair([]) do
+    {:error, "No participants, no secret santas!"}
+  end
+
+  def pair([_]) do
+    {:error, "Alone for Christmas day? Watch 'Home Alone' 1 and 2!"}
+  end
+
+  def pair([_ | _] = participants) do
+    {:ok,
+     participants
+     |> Enum.shuffle()
+     |> EnumUtils.rotate_left()
+     |> Enum.zip(participants)
```

Comment:
It seems to me there's no guarantee you won't end up with a person giving to themselves. Is that the case ?

Reply:
Indeed, the zip should be done with the already shuffled participants.

Reply:
Good catch! I'll fix it.